ui/js/controllers/jobs.js

```diff
@@ -177,19 +177,38 @@ treeherderApp.controller('ResultSetCtrl', [
             $rootScope.$emit(thEvents.deleteRunnableJobs, $scope.resultset);
         };
 
-        $scope.cancelAllJobs = function(revision) {
-            if (!window.confirm('This will cancel all pending and running jobs for revision ' + revision + '!\n\nClick "OK" if you\'re sure.')) {
+        $scope.cancelJobs = function(revision) {
+            var job = ThResultSetStore.getSelectedJob($scope.repoName).job;
+            var single_job = job !== undefined;
+            var message = single_job ?
```

Comment:
Actually I changed my mind, it probably is a good idea. But let's call the variable `singleJobSelected` to make things more clear.